compiler/pre_hook_amd-llvm.cmake

```
@@ -35,6 +35,13 @@ else()
     # For now we will switch the priorty for find_package to first search in
     # CONFIG mode.
     set(RUNTIMES_CMAKE_ARGS "-DCMAKE_FIND_PACKAGE_PREFER_CONFIG=ON")
+
+    # We may need to guard this for amd-staging only builds.
+    if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/../openmp/device/CMakeLists.txt")
```

NIT: Elsewhere `THEROCK_SOURCE_DIR` is used. Could do here as well but no strict need to.

```suggestion
    if(EXISTS "${THEROCK_SOURCE_DIR}/compiler/amd-llvm/openmp/device/CMakeLists.txt")
```